lib/ansible/modules/remote_management/spacewalk/spacewalk_swchannel_facts.py

```diff
@@ -0,0 +1,73 @@
+#!/usr/bin/python
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+# Copyright: (c) 2018, Davide Blasi (@davegarath)
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+ANSIBLE_METADATA = {
+    'metadata_version': '1.1',
+    'status': ['preview'],
+    'supported_by': 'community'
+}
+
+DOCUMENTATION = '''
+---
+module: spacewalk_swchannel_facts
+short_description: Gather facts about all channels in spacewalk
+description:
+  - "Gather facts about all channels in spacewalk"
+version_added: "2.8"
+author:
+  - Davide Blasi (@davegarath)
+extends_documentation_fragment: spacewalk.documentation
+'''
+
+EXAMPLES = '''
+- name: Get all channels in spacewalk
+  spacewalk_swchannel_facts:
+      url: "{{ spacewalk_url_api }}"
+      login: "{{ spacewalk_login }}"
+      password: "{{ spacewalk_passowrd }}"
+  delegate_to: localhost
+  register: channels
+
+- debug: var=channels
+'''
+
+RETURN = '''
+channels_facts:
+  description: Provides all spacewalk channels in ansible_facts
+  type: dict
+  returned: always
+  sample:
+    "spacewalk_channels": {
```

Comment:
The formatting for Ansible's documentation on examples like this isn't optimal as it merges everything into a single line. My preference is to have a type: complex and detail each returned value if possible. Not a major issue in my opinion as many modules use this format. See the `meraki_network` module in `network/` if you need an example.

Reply:
rewritten